<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Rashika W. Ranasinghe" />


<title>Genetic and Biochemical Foundations of Carotenoid Coloration in Flameback Woodpeckers</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Data Visualization</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Genetic and Biochemical Foundations of
Carotenoid Coloration in Flameback Woodpeckers</h1>
<h4 class="author">Rashika W. Ranasinghe</h4>
<h4 class="date">1/30/2025</h4>

</div>


<p><strong>Cite as:</strong></p>
<p>Rashika W. Ranasinghe, Jocelyn Hudon, Sampath Seneviratne, et
al. Biochemical and Genomic Underpinnings of Carotenoid Color Variation
across a Hybrid Zone between South Asian Flameback Woodpeckers.
Authorea. January 09, 2025. <a
href="DOI:10.22541/au.173643127.76349831/v1"
class="uri">DOI:10.22541/au.173643127.76349831/v1</a></p>
<p><br> Welcome! This website offers step-by-step instructions for
reproducing the data analysis and visualizations presented in our
paper:</p>
<p><strong>“Biochemical and Genomic Underpinnings of Carotenoid Color
Variation Across a Hybrid Zone Between South Asian Flameback
Woodpeckers.”</strong> <br></p>
<p><br> Load the libraries:</p>
<pre class="r"><code>library(pavo)
library(stringr)
library(ggplot2)
library(patchwork)</code></pre>
<pre><code>## Warning: package &#39;patchwork&#39; was built under R version 4.4.1</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ forcats   1.0.0     ✔ readr     2.1.5
## ✔ lubridate 1.9.3     ✔ tibble    3.2.1
## ✔ purrr     1.0.2     ✔ tidyr     1.3.1</code></pre>
<pre><code>## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(qqman)</code></pre>
<pre><code>## 
## For example usage please run: vignette(&#39;qqman&#39;)
## 
## Citation appreciated but not required:
## Turner, (2018). qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots. Journal of Open Source Software, 3(25), 731, https://doi.org/10.21105/joss.00731.</code></pre>
<div id="main-figures" class="section level3">
<h3>Main Figures</h3>
<div id="figure-2a-reflectance-curves-for-crown-feathers"
class="section level4">
<h4>Figure 2a: Reflectance curves for crown feathers</h4>
<p>This section demonstrates how to create reflectance curves for crown
feathers across the three populations.</p>
<p>Load the Data: The reflectance data (in .txt format) for each
individual’s crown feathers are stored in the “Dinopium_crown/” folder.
Each feather has three replicate readings.</p>
<pre class="r"><code># Read the data
crown_reflect_Data &lt;- getspec(&quot;Data/Dinopium_crown/&quot;, ext = &quot;txt&quot;, decimal = &quot;.&quot;)</code></pre>
<pre><code>## 123 files found; importing spectra:</code></pre>
<p>Quickly visualize the raw data.</p>
<pre class="r"><code>plot(crown_reflect_Data)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<p>Replace any negative reflectance values with zero and re-plot to
confirm the changes.</p>
<pre class="r"><code>crown_nonNeg &lt;- procspec(crown_reflect_Data, fixneg = &quot;zero&quot;) </code></pre>
<pre><code>## processing options applied:
## Negative value correction: converted negative values to zero</code></pre>
<pre class="r"><code>plot(crown_nonNeg)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>Calculate the average reflectance for each sample across the three
replicates.</p>
<pre class="r"><code>crown_mean &lt;- aggspec(crown_nonNeg, by = 3, FUN = mean)</code></pre>
<p>Create a grouping variable to separate the spectra by population, and
then plot the curves with error estimates and annotations for the hue
values (mean ± 95% CI) for each phenotype group.</p>
<pre class="r"><code># Create a grouping variable that separates the spectra by the phenotype
crown_pop &lt;- gsub(&quot;^[^_]*_([^_]+)_.*$&quot;, &quot;\\1&quot;, names(crown_mean))[-1]
crown_pheno &lt;- str_replace_all(crown_pop, c(&quot;Allo.R&quot;=&quot;D.psarodes&quot;, 
                                            &quot;SymR&quot;=&quot;D.psarodes&quot;, 
                                            &quot;Jaffna&quot;=&quot;D.benghalense&quot;, 
                                            &quot;Mannar&quot;=&quot;D.benghalense&quot;, 
                                            &quot;SymY&quot;=&quot;D.benghalense&quot;, 
                                            &quot;HYB&quot;=&quot;Hybrids&quot;))

# Make the plot
aggplot(crown_mean, crown_pheno, FUN.error=function(x)1.96*sd(x)/sqrt(length(x)), 
        lcol=c(&quot;red&quot;, &quot;green3&quot;,&quot;orange&quot;),
        shadecol=c(&quot;red&quot;, &quot;green3&quot;,&quot;orange&quot;),
        xlim=c(375, 700),ylim=c(0, 35),alpha=0.1,lwd=1.5, legend=T, main=&quot;Crown Feathers&quot;)

# Annotate with lambda R50 mean and 95% CI for each group
abline(v=600, lty=2, col=&quot;red&quot;) 
rect(597, 0, 603, 40, border = &quot;transparent&quot;, col = rgb(1, 0, 0, alpha = 0.09))
abline(v=598, lty=2, col=&quot;green3&quot;) 
rect(596, 0, 619, 40, border = &quot;transparent&quot;, col = rgb(0, 1, 0, alpha = 0.09))
abline(v=598, lty=2, col=&quot;orange&quot;) 
rect(578, 0, 619, 40, border = &quot;transparent&quot;, col = rgb(1, 0.5, 0, alpha = 0.09))</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="figure-2e-reflectance-curves-for-mantle-feathers"
class="section level4">
<h4>Figure 2e: Reflectance curves for mantle feathers</h4>
<p>The method for generating reflectance curves for mantle feathers is
the same as that used for crown feathers</p>
<p>Load the data</p>
<pre class="r"><code>mantle_reflect_Data &lt;- getspec(&quot;Data/Dinopium_mantle/&quot;, ext = &quot;txt&quot;, decimal = &quot;.&quot;)</code></pre>
<pre><code>## 123 files found; importing spectra:</code></pre>
<p>Replace negative values with zeros.</p>
<pre class="r"><code>mantle_reflect_Data_nonNeg &lt;- procspec(mantle_reflect_Data, fixneg = &quot;zero&quot;)</code></pre>
<pre><code>## processing options applied:
## Negative value correction: converted negative values to zero</code></pre>
<p>Compute the average across the three replicate readings.</p>
<pre class="r"><code>mantle_mean &lt;- aggspec(mantle_reflect_Data_nonNeg, by = 3, FUN = mean)</code></pre>
<p>Create a phenotype grouping variable and generate the reflectance
curves with appropriate annotations.</p>
<pre class="r"><code>mantle_pop &lt;- gsub(&quot;^[^_]*_([^_]+)_.*$&quot;, &quot;\\1&quot;, names(mantle_mean))[-1] # to create a grouping variable that separates the spectra by the population
mantle_pheno &lt;- str_replace_all(mantle_pop, c(&quot;Allo.R&quot;=&quot;D.psarodes&quot;, 
                                              &quot;SymR&quot;=&quot;D.psarodes&quot;, 
                                              &quot;Jaffna&quot;=&quot;D.benghalense&quot;, 
                                              &quot;Mannar&quot;=&quot;D.benghalense&quot;, 
                                              &quot;SymY&quot;=&quot;D.benghalense&quot;, 
                                              &quot;HYB&quot;=&quot;Hybrids&quot;))

# Make the plot
aggplot(mantle_mean, mantle_pheno, FUN.error=function(x)1.96*sd(x)/sqrt(length(x)), 
        lcol=c(&quot;red&quot;, &quot;green3&quot;,&quot;orange&quot;),
        shadecol=c(&quot;red&quot;, &quot;green3&quot;,&quot;orange&quot;),
        xlim=c(375, 700),alpha=0.1,legend=TRUE, main=&quot;Mantle reflectance among Dinopium phenotype groups&quot;)

# Annotate with lambda R50 mean and 95% CI for each phenotype group
abline(v=602, lty=2, col=&quot;red&quot;) 
rect(599, 0, 604, 40, border = &quot;transparent&quot;, col = rgb(1, 0, 0, alpha = 0.09))
abline(v=546, lty=2, col=&quot;green3&quot;) 
rect(541, 0, 551, 40, border = &quot;transparent&quot;, col = rgb(0, 1, 0, alpha = 0.09))
abline(v=575, lty=2, col=&quot;orange&quot;) 
rect(567, 0, 583, 40, border = &quot;transparent&quot;, col = rgb(1, 0.5, 0, alpha = 0.09))</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p><br></p>
</div>
<div id="figure-2b-d-and-2f-h-visualization-of-colorimatric-parameters"
class="section level4">
<h4>Figure 2b-d and 2f-h: Visualization of Colorimatric Parameters</h4>
<p>These figures present three key colorimetric parameters: brightness,
carotenoid chroma, and hue, derived from the reflectance data. The
parameters are visualized separately for crown feathers (Figures 2b–d)
and mantle feathers (Figures 2f–h). These parameters were estimated
using the <em>pavo</em> package in R.</p>
<p>Calculate the colorimatric variables and save them as a .csv file</p>
<pre class="r"><code># For crown feathers
crown_colormatrics &lt;- summary(crown_mean)
# To save the data
write.csv(crown_colormatrics, file = &quot;Data/Colorimatric.Data-Dinopium_Crown.csv&quot;)

# For mantle feathers
mantle_colormatrics &lt;- summary(mantle_mean)
# To save the data
write.csv(mantle_colormatrics, file = &quot;Data/Colorimatric.Data-Dinopium_Mantle.csv&quot;)</code></pre>
<p>Please note that Ranasinghe et al., 2025 have plotted the hue values
(H3) that were manually calculated from the reflectance data, rather
than the H3 values generated by <em>pavo</em>. Load the updated data
files with the new hue (H3) values and generate the plots.</p>
<pre class="r"><code># Read the edited mantle data
colorimatrics &lt;- read.csv(&quot;Data/Colorimatric.Data-Dinopium mantle and crown from pavo.csv&quot;, stringsAsFactors = T)

# Reshape the data 
colorimatrics$Species &lt;- factor(colorimatrics$Species, levels = c(&quot;D. benghalense&quot;, &quot;Hybrid&quot;,  &quot;D. psarodes&quot;))</code></pre>
<p><br> ##### Figure 2b and 2f: Brightness</p>
<pre class="r"><code>ggplot(colorimatrics, aes(y = B2, x = Species, fill=Species)) +
  geom_jitter(position = position_jitter(0.3), size = 3.5, shape = 21, color = &quot;black&quot;, alpha = 0.6) +
  stat_summary(fun.data = &quot;mean_cl_normal&quot;,  fun.args = list(mult = 2.5), 
               geom = &quot;pointrange&quot;,  size = 0.5, shape = 23, fill = &quot;black&quot;)+
  facet_wrap(~Feather, ncol =2) +
  scale_fill_manual(values = c(&quot;yellow&quot;, &quot;orange&quot;, &quot;red&quot;),
                    name = NULL,
                    labels = c(expression(paste(&quot;Yellow &quot;, italic(&quot;(D. benghalense)&quot;))), 
                                &quot;Intermediate phenotype&quot;, 
                               expression(paste(&quot;Red &quot;, italic(&quot;(D. psarodes)&quot;))))) +
   scale_y_continuous(breaks = seq(5, 25, by = 5), limits = c(5, 25)) +
  ylab(&quot;Brightness&quot;) +
  xlab(&quot;Phenotypic groups&quot;)+
  scale_x_discrete(labels = NULL) +
  theme_linedraw() +
  theme(
    aspect.ratio = 1,
    panel.grid.major = element_line(color = &quot;white&quot;),
    panel.grid.minor = element_line(color = &quot;white&quot;),
    strip.background = element_rect(color = &quot;grey50&quot;, fill = &quot;grey50&quot;),  
    strip.text = element_text(face = &quot;bold&quot;, size = 11,
                              margin = margin(10, 10, 10, 10)),
    legend.text = element_text(size = 11),
    panel.spacing = unit(1.5, &quot;lines&quot;),
    legend.position = &quot;top&quot;,
    axis.title = element_text(size = 15, face = &quot;bold&quot;),
    axis.text = element_text(size = 14))</code></pre>
<pre><code>## Warning: Removed 2 rows containing non-finite outside the scale range
## (`stat_summary()`).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p><br> ##### Figure 2c and 2g: Caroatenoid Chroma</p>
<pre class="r"><code>ggplot(colorimatrics, aes(y = S9, x = Species, fill=Species)) +
  geom_jitter(position = position_jitter(0.3), size = 3.5, shape = 21, color = &quot;black&quot;, alpha = 0.6) +
  stat_summary(fun.data = &quot;mean_cl_normal&quot;,  fun.args = list(mult = 2.5), 
               geom = &quot;pointrange&quot;,  size = 0.5, shape = 23, fill = &quot;black&quot;)+
  facet_wrap(~Feather, ncol =2) +
  scale_fill_manual(values = c(&quot;yellow&quot;, &quot;orange&quot;, &quot;red&quot;),
                    name = NULL,
                    labels = c(expression(paste(&quot;Yellow &quot;, italic(&quot;(D. benghalense)&quot;))), 
                                &quot;Intermediate phenotype&quot;, 
                               expression(paste(&quot;Red &quot;, italic(&quot;(D. psarodes)&quot;))))) +
    scale_y_continuous(breaks = seq(0, 1, by = 0.2), limits = c(0, 1)) +
  ylab(&quot;Carotenoid chroma&quot;) +
  xlab(&quot;Phenotypic groups&quot;)+
  scale_x_discrete(labels = NULL) +
  theme_linedraw() +
  theme(
    aspect.ratio = 1,
    panel.grid.major = element_line(color = &quot;white&quot;),
    panel.grid.minor = element_line(color = &quot;white&quot;),
    strip.background = element_rect(color = &quot;grey50&quot;, fill = &quot;grey50&quot;),  
    strip.text = element_text(face = &quot;bold&quot;, size = 11,
                              margin = margin(10, 10, 10, 10)),
    legend.text = element_text(size = 11),
    panel.spacing = unit(1.5, &quot;lines&quot;),
    legend.position = &quot;top&quot;,
    axis.title = element_text(size = 15, face = &quot;bold&quot;),
    axis.text = element_text(size = 14))</code></pre>
<pre><code>## Warning: Removed 1 row containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p><br> ##### Figure 2d and 2h: Hue</p>
<pre class="r"><code>ggplot(colorimatrics, aes(y = H3, x = Species, fill=Species)) +
  geom_jitter(position = position_jitter(0.3), size = 3.5, shape = 21, color = &quot;black&quot;, alpha = 0.6) +
  stat_summary(fun.data = &quot;mean_cl_normal&quot;,  fun.args = list(mult = 2.5), 
               geom = &quot;pointrange&quot;,  size = 0.5, shape = 23, fill = &quot;black&quot;)+
  facet_wrap(~Feather, ncol =2) +
  scale_fill_manual(values = c(&quot;yellow&quot;, &quot;orange&quot;, &quot;red&quot;),
                    name = NULL,
                    labels = c(expression(paste(&quot;Yellow &quot;, italic(&quot;(D. benghalense)&quot;))), 
                                &quot;Intermediate phenotype&quot;, 
                               expression(paste(&quot;Red &quot;, italic(&quot;(D. psarodes)&quot;))))) +
    scale_y_continuous(breaks = seq(530, 610, by = 20), limits = c(530, 610)) +
  ylab(&quot;Hue (nm)&quot;) +
  xlab(&quot;Phenotypic groups&quot;)+
  scale_x_discrete(labels = NULL) +
  theme_linedraw() +
  theme(
    aspect.ratio = 1,
    panel.grid.major = element_line(color = &quot;white&quot;),
    panel.grid.minor = element_line(color = &quot;white&quot;),
    strip.background = element_rect(color = &quot;grey50&quot;, fill = &quot;grey50&quot;),  
    strip.text = element_text(face = &quot;bold&quot;, size = 11,
                              margin = margin(10, 10, 10, 10)),
    legend.text = element_text(size = 11),
    panel.spacing = unit(1.5, &quot;lines&quot;),
    legend.position = &quot;top&quot;,
    axis.title = element_text(size = 15, face = &quot;bold&quot;),
    axis.text = element_text(size = 14))</code></pre>
<pre><code>## Warning: Removed 5 rows containing non-finite outside the scale range
## (`stat_summary()`).</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
</div>
<div id="figure-3a-carotenoid-pigment-concentrations"
class="section level4">
<h4>Figure 3a: Carotenoid pigment concentrations</h4>
<p>This figure shows the pigment concentrations identified through
high-performance liquid chromatography (HPLC) analysis.</p>
<pre class="r"><code># Read the data
HPLC &lt;- read.csv(&quot;Data/HPLC_data_Mantle_Crown_feathers_All_Diniopium.csv&quot;, stringsAsFactors = T)

# Reshape the data
Pigment_content &lt;- HPLC %&gt;%
  pivot_longer(cols = c(lutein.ident__concentration:papilioerythrinone_concentration),
               names_to = &quot;Variable&quot;,
               values_to = &quot;Values&quot;)
levels(as.factor(Pigment_content$Variable))</code></pre>
<pre><code>##  [1] &quot;adonirubin_concentration&quot;               
##  [2] &quot;alpha..doradexanthin_concentration&quot;     
##  [3] &quot;astaxanthin_concentration&quot;              
##  [4] &quot;beta.cryptoxanthin_concentration&quot;       
##  [5] &quot;canary.xanthophyll.A_concentration&quot;     
##  [6] &quot;canary.xanthophyll.B_concentration&quot;     
##  [7] &quot;canthaxanthin_concentration&quot;            
##  [8] &quot;lutein.ident__concentration&quot;            
##  [9] &quot;papilioerythrinone_concentration&quot;       
## [10] &quot;X3..dehydro.lutein.ident__concentration&quot;
## [11] &quot;zeaxanthin._concentration&quot;</code></pre>
<pre class="r"><code>Pigment_content$Variable &lt;- factor(Pigment_content$Variable, levels = c(&quot;beta.cryptoxanthin_concentration&quot;, &quot;zeaxanthin._concentration&quot;, &quot;lutein.ident__concentration&quot;, &quot;X3..dehydro.lutein.ident__concentration&quot;, &quot;canary.xanthophyll.A_concentration&quot;, &quot;canary.xanthophyll.B_concentration&quot;, &quot;alpha..doradexanthin_concentration&quot;, &quot;papilioerythrinone_concentration&quot;, &quot;canthaxanthin_concentration&quot;,&quot;adonirubin_concentration&quot;,&quot;astaxanthin_concentration&quot;))

# Make the plot
ggplot(Pigment_content, aes(fill = Variable, y = Values, x = Variable)) +
  geom_jitter(position = position_jitter(0.3), size = 2.3, shape = 21, color = &quot;black&quot;, alpha = 0.6) +
  stat_summary(fun.data = &quot;mean_cl_normal&quot;,  fun.args = list(mult = 1), 
               geom = &quot;pointrange&quot;,  size = 0.4, shape = 23, fill = &quot;black&quot;)+
  facet_wrap(~feathers + Scientific.name, 
             labeller = labeller(Scientific.name = c(&quot;Dinopium benghalense jaffnense&quot; = &quot;D.benghalense&quot;,
                                                     &quot;Dinopium hybrid &quot; = &quot;Hybrids&quot;,
                                                     &quot;Dinopium psarodes  &quot; = &quot;D. psarodes&quot;), size=0.5)) + 
  scale_fill_manual(values = c(&quot;yellow&quot;, &quot;yellow&quot;, &quot;yellow&quot;, &quot;gold3&quot;, &quot;gold3&quot;, &quot;gold3&quot;, &quot;red3&quot;, &quot;red3&quot;, &quot;red3&quot;, &quot;red3&quot;, &quot;red3&quot;),
                    labels = c(&quot;beta.cryptoxanthin_concentration&quot;=&quot;beta-cryptoxanthin&quot;,
                               &quot;zeaxanthin._concentration&quot;=&quot;Zeaxanthin&quot;,
                               &quot;lutein.ident__concentration&quot;=&quot;Lutein&quot;, 
                               &quot;X3..dehydro.lutein.ident__concentration&quot;=&quot;3&#39;-dehydrolutein&quot;,
                               &quot;canary.xanthophyll.A_concentration&quot;=&quot;Canary xanthophyll A&quot;,
                               &quot;canary.xanthophyll.B_concentration&quot;=&quot;Canary xanthophyll B&quot;,
                               &quot;canthaxanthin_concentration&quot;=&quot;Canthaxanthin&quot;,
                               &quot;adonirubin_concentration&quot;=&quot;Adonirubin&quot;,
                               &quot;astaxanthin_concentration&quot;=&quot;Astaxanthin&quot;,  
                               &quot;alpha..doradexanthin_concentration&quot;=&quot;alpha-doradexanthin&quot;,
                               &quot;papilioerythrinone_concentration&quot;=&quot;Papilioerythrinone&quot;), name = NULL) +
  xlab(&quot;Carotenoid pigments&quot;) + 
  ylab(&quot;Concentration (mg/g)&quot;) + 
  scale_x_discrete(labels =c(&quot;beta.cryptoxanthin_concentration&quot; = &quot;β-Cryptoxanthin&quot;,
                              &quot;zeaxanthin._concentration&quot; = &quot;Zeaxanthin&quot;,
                              &quot;lutein.ident__concentration&quot; = &quot;Lutein&quot;, 
                              &quot;X3..dehydro.lutein.ident__concentration&quot; = &quot;3&#39;-Dehydrolutein&quot;,
                              &quot;canary.xanthophyll.A_concentration&quot; = &quot;Canary Xanthophyll A&quot;,
                              &quot;canary.xanthophyll.B_concentration&quot; = &quot;Canary Xanthophyll B&quot;,
                              &quot;canthaxanthin_concentration&quot; = &quot;Canthaxanthin&quot;,
                              &quot;adonirubin_concentration&quot; = &quot;Adonirubin&quot;,
                              &quot;astaxanthin_concentration&quot; = &quot;Astaxanthin&quot;,  
                              &quot;alpha..doradexanthin_concentration&quot; = &quot;α-Doradexanthin&quot;,
                              &quot;papilioerythrinone_concentration&quot; = &quot;Papilioerythrinone&quot;)) +
  theme_linedraw() +
  theme(
    panel.grid.major = element_line(color = &quot;grey90&quot;),
    panel.grid.minor = element_line(color = &quot;grey80&quot;),
    strip.background = element_rect(color = &quot;grey50&quot;, fill = &quot;grey50&quot;, size = 0.1),
    strip.text = element_text(face = &quot;bold&quot;, color = &quot;white&quot;),
    axis.title.y = element_text(face = &quot;bold&quot;),
    axis.title.x = element_text(face = &quot;bold&quot;, margin = margin(0, 0, 30, 0)),
    strip.text.x = element_text(margin = margin(0.05,0,0.05,0, &quot;cm&quot;)),
    legend.position = &quot;none&quot;,
    legend.title = element_blank(),
    axis.text.x = element_text(angle = 50, hjust = 1)) </code></pre>
<pre><code>## Warning: The `size` argument of `element_rect()` is deprecated as of ggplot2 3.4.0.
## ℹ Please use the `linewidth` argument instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
<div
id="figure-3b-carotenoid-composition-in-mantle-feathers-of-intermediates"
class="section level4">
<h4>Figure 3b: Carotenoid composition in mantle feathers of
intermediates</h4>
<p>This figure presents the absolute concentration and composition of
red and yellow carotenoids in the mantle feathers of intermediate
orange-backed flamebacks, including individuals with multiple feather
samples. The scatter plots at the top show the hue for each feather.</p>
<pre class="r"><code># Filter the data to subset only the hybrids
HPLC_HYB_mantle &lt;- filter(HPLC, Scientific.name ==&quot;Dinopium hybrid &quot; &amp; feathers == &quot;Mantle&quot;) 

# Reshape the data 
Pigment_content &lt;- HPLC_HYB_mantle %&gt;%
  pivot_longer(cols = c(lutein.ident__concentration:papilioerythrinone_concentration),
               names_to = &quot;Variable&quot;,
               values_to = &quot;Values&quot;)

Pigment_content$Variable &lt;- factor(Pigment_content$Variable, levels = c(&quot;beta.cryptoxanthin_concentration&quot;, &quot;zeaxanthin._concentration&quot;, &quot;lutein.ident__concentration&quot;, &quot;X3..dehydro.lutein.ident__concentration&quot;, &quot;canary.xanthophyll.A_concentration&quot;, &quot;canary.xanthophyll.B_concentration&quot;, &quot;canthaxanthin_concentration&quot;,&quot;adonirubin_concentration&quot;,&quot;astaxanthin_concentration&quot;,  &quot;alpha..doradexanthin_concentration&quot;, &quot;papilioerythrinone_concentration&quot;))

Pigment_content$Identification_II &lt;- factor(Pigment_content$Identification_II, levels = c(&quot;UD23RR01-1&quot;,&quot;UD23RR01-2&quot;,&quot;UE07RR01-1&quot;, &quot;UE07RR01-2&quot;,&quot;UC20RR08&quot;,&quot;UE07RR03&quot;, &quot;UC20RR04-1&quot;, &quot;UC20RR04-2&quot;, &quot;UC20RR04-3&quot;, &quot;UC30RR01-1&quot;,&quot;UC30RR01-2&quot;, &quot;UC29RR04&quot;, &quot;UC28RR02&quot;))

# Plot the carotenoid contents
p1 &lt;- ggplot(Pigment_content, aes(x=Identification_II, y=Values, fill=Variable))+
  geom_bar(stat = &quot;summary&quot;, fun = &quot;mean&quot;, color=&quot;black&quot;, alpha=0.85)+
  labs(x=&quot;Hybrid Sample ID&quot;, y= &quot;Pigment concentration (mg/g)&quot;)+
  scale_fill_manual(values = c(&quot;lutein.ident__concentration&quot;=&quot;yellow&quot;,
                               &quot;zeaxanthin._concentration&quot;=&quot;yellow3&quot;,
                               &quot;beta.cryptoxanthin_concentration&quot;=&quot;yellow4&quot;,
                               &quot;X3..dehydro.lutein.ident__concentration&quot;=&quot;gold&quot;,
                               &quot;canary.xanthophyll.A_concentration&quot;=&quot;gold3&quot;,
                               &quot;canary.xanthophyll.B_concentration&quot;=&quot;lightgoldenrod1&quot;,
                               &quot;canthaxanthin_concentration&quot; = &quot;orangered&quot;,
                               &quot;astaxanthin_concentration&quot; = &quot;red&quot;,
                               &quot;adonirubin_concentration&quot; = &quot;indianred&quot;,
                               &quot;alpha..doradexanthin_concentration&quot; = &quot;darkred&quot;,
                               &quot;papilioerythrinone_concentration&quot;=&quot;palevioletred3&quot;),
                    labels=c(&quot;lutein.ident__concentration&quot;=&quot;Lutein&quot;,
                             &quot;zeaxanthin._concentration&quot;=&quot;Zeaxanthin&quot;,
                             &quot;beta.cryptoxanthin_concentration&quot;=&quot;beta-cryptoxanthin&quot;,
                             &quot;X3..dehydro.lutein.ident__concentration&quot;=&quot;3&#39;-dehydrolutein&quot;,
                             &quot;canary.xanthophyll.A_concentration&quot;=&quot;Canary xanthophyll A&quot;,
                             &quot;canary.xanthophyll.B_concentration&quot;=&quot;Canary xanthophyll B&quot;,
                             &quot;canthaxanthin_concentration&quot; = &quot;Canthaxanthin&quot;,
                             &quot;astaxanthin_concentration&quot; = &quot;Astaxanthin&quot;,
                             &quot;adonirubin_concentration&quot; = &quot;Adonirubin&quot;,
                             &quot;alpha..doradexanthin_concentration&quot; = &quot;Alpha-doradexanthin&quot;,
                             &quot;papilioerythrinone_concentration&quot;=&quot;Papilioerythrinone&quot;)) +
  theme_linedraw()+
  theme(
    plot.title = element_text(size = 12),
    axis.text.x = element_text(size=9, angle = 20, hjust = 1),
    axis.text.y = element_text(size=13),
    axis.title = element_text(size=12),
    panel.grid.major = element_line(color = &quot;grey30&quot;),
    panel.grid.minor = element_line(color = &quot;white&quot;),
    legend.position = &quot;right&quot;,
    legend.text = element_text(size = 10),
    legend.title = element_blank())+guides(fill = guide_legend(nrow = 11))

# Display hue of each sample
HPLC_HYB_mantle$Identification_II &lt;- factor(HPLC_HYB_mantle$Identification_II, levels = c(&quot;UD23RR01-1&quot;,&quot;UD23RR01-2&quot;,&quot;UE07RR01-1&quot;, &quot;UE07RR01-2&quot;,&quot;UC20RR08&quot;,&quot;UE07RR03&quot;, &quot;UC20RR04-1&quot;, &quot;UC20RR04-2&quot;, &quot;UC20RR04-3&quot;, &quot;UC30RR01-1&quot;,&quot;UC30RR01-2&quot;, &quot;UC29RR04&quot;, &quot;UC28RR02&quot;))

p2 &lt;- ggplot(HPLC_HYB_mantle, aes(y = Reflectance.average.Lambda.R50, x = Identification_II)) + 
  geom_point(shape = 21, fill = &quot;grey80&quot;, color = &quot;black&quot;, size = 2.3, stroke = 0.5) + 
  scale_x_discrete(labels =NULL)+
  ylim(c(545, 600)) +
  labs(y = &quot;Hue (nm)&quot;, x = &quot;&quot;) +
  theme_linedraw() +
  theme(axis.title.x = element_blank(),
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank())

# Make the both plots
p2 / p1 + plot_layout(heights = c(0.25, 1))</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p><br> #### Figure 4a and f: Frequency of functional groups for crown
and mantle feathers</p>
<pre class="r"><code># Read the data
HPLC &lt;- read.csv(&quot;Data/HPLC_data_Mantle_Crown_feathers_All_Diniopium.csv&quot;, stringsAsFactors = T)

# Manipulate the data
HPLC_long &lt;- HPLC %&gt;%
  pivot_longer(cols = c(avg.4.keto_groups:avg.3.oxygenation,avg.epsilon.rings),
               names_to = &quot;Variable&quot;,
               values_to = &quot;Values&quot;)
levels(as.factor(HPLC_long$Variable))</code></pre>
<pre><code>## [1] &quot;avg.3.oxygenation&quot; &quot;avg.4.keto_groups&quot; &quot;avg.epsilon.rings&quot;</code></pre>
<pre class="r"><code>HPLC_long$Variable &lt;- factor(HPLC_long$Variable, levels = c(&quot;avg.3.oxygenation&quot;, &quot;avg.epsilon.rings&quot;, &quot;avg.4.keto_groups&quot;))


ggplot(HPLC_long, aes(y = Values, x = Scientific.name, fill=Scientific.name)) +
  geom_jitter(position = position_jitter(0.3), size = 2, shape = 21, color = &quot;black&quot;, alpha = 0.6) +
  stat_summary(fun.data = &quot;mean_cl_normal&quot;,  fun.args = list(mult = 2.5), 
               geom = &quot;pointrange&quot;,  size = 0.3, shape = 23, fill = &quot;black&quot;)+
  facet_wrap(~feathers + Variable, ncol = 3, scales = &quot;fixed&quot;, 
             labeller = labeller(Variable = c(&quot;avg.epsilon.rings&quot;=&quot;Epsilon rings&quot;,
                                              &quot;avg.3.oxygenation&quot;=&quot;C3(3&#39;)-Oxygenated groups&quot;,
                                              &quot;avg.4.keto_groups&quot;=&quot;C4(4&#39;)-keto groups&quot;))) +
  scale_fill_manual(values = c(&quot;yellow&quot;, &quot;orange&quot;, &quot;red&quot;), 
                    name = NULL,
                    labels = c(expression(paste(&quot;Yellow &quot;, italic(&quot;(D. benghalense)&quot;))), &quot;Hybrids&quot;,
                               expression(paste(&quot;Red &quot;, italic(&quot;(D. psarodes)&quot;))))) +
  scale_x_discrete(labels =c(&quot;Dinopium benghalense jaffnense&quot; =  expression(italic(&quot;D. benghalense&quot;)),
                             &quot;Dinopium hybrid &quot; = &quot;Intermediates&quot;,
                             &quot;Dinopium psarodes  &quot;=expression(italic(&quot;D. psarodes&quot;)))) +
  ylab(&quot;Average number of functional groups&quot;) +
  xlab(&quot;Phenotypic groups&quot;)+
  theme_linedraw() +
  theme(
    panel.grid.major = element_line(color = &quot;grey90&quot;),
    panel.grid.minor = element_line(color = &quot;white&quot;),
    strip.background = element_rect(color = &quot;grey50&quot;, fill = &quot;grey50&quot;),  
    strip.text = element_text(face = &quot;bold&quot;, color = &quot;white&quot;),
    panel.spacing = unit(0.5, &quot;lines&quot;),
    legend.position = &quot;top&quot;,
    axis.title.y = element_text(face = &quot;bold&quot;),
    axis.title.x = element_text(face = &quot;bold&quot;),
    axis.text.x = element_text(size=9, angle = 20, hjust = 1),
    strip.text.x = element_text(margin = margin(0.001,0,0.01,0, &quot;cm&quot;))) </code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
</div>
<div
id="figure-4g---i-correlation-between-frequency-of-functional-groups-and-hue"
class="section level4">
<h4>Figure 4g - i: Correlation between frequency of functional groups
and hue</h4>
<p>The correlation analysis between frequency of functional groups and
hue was done only for mantle feathers. Therefore subset only the mantle
feather data.</p>
<pre class="r"><code># Read the data
HPLC &lt;- read.csv(&quot;Data/HPLC_data_Mantle_Crown_feathers_All_Diniopium.csv&quot;, stringsAsFactors = T)
HPLC_mantle &lt;- filter(HPLC, feathers == &quot;Mantle&quot;)</code></pre>
<p>Perform the correlation analysis between each functional group
variable (number of C3(3’)-oxygenated groups, number of ε-end rings, and
number of C4(4’)-keto groups) and mantle hue before generating the
figure.</p>
<div id="generate-figure-4g" class="section level5">
<h5>Generate figure 4g:</h5>
<pre class="r"><code># Spearman&#39;s rank correlation between average number of C3(3’)-oxygenated groups mantle hue:
cor.test(HPLC_mantle$Reflectance.average.Lambda.R50, HPLC_mantle$avg.3.oxygenation, method = &quot;spearman&quot;)</code></pre>
<pre><code>## 
##  Spearman&#39;s rank correlation rho
## 
## data:  HPLC_mantle$Reflectance.average.Lambda.R50 and HPLC_mantle$avg.3.oxygenation
## S = 23140, p-value = 0.0002679
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##        rho 
## -0.5243742</code></pre>
<pre class="r"><code># Make the Figure
ggplot(HPLC_mantle, aes(x = Reflectance.average.Lambda.R50, y = avg.3.oxygenation)) +
  geom_point(aes(fill = Scientific.name), size = 2, shape = 21, color = &quot;black&quot;, alpha = 0.6) +
  geom_smooth(method = &quot;lm&quot;, se = F, color = &quot;black&quot;, size=0.5, linetype = &quot;dashed&quot;) + 
  scale_fill_manual(values = c(&quot;yellow&quot;, &quot;orange&quot;, &quot;red&quot;), name = NULL,
                    labels = c(expression(italic(&quot;D. benghalense&quot;)), &quot;Intermediates&quot;, expression(italic(&quot;D. psarodes&quot;)))) +
  xlab(&quot;Mantle hue (nm)&quot;) + 
  ylab(&quot;C3(3&#39;)-oxygenated groups&quot;) +
  ylim(c(0, 2.5))+
  theme_linedraw() +
  theme(
    panel.grid.major = element_line(color = &quot;grey90&quot;),
    panel.grid.minor = element_line(color = &quot;white&quot;),
    legend.position = &quot;top&quot;,
    plot.title = element_text(size = 8, face = &quot;bold&quot;)) + 
  annotate(&quot;text&quot;, x = Inf, y = Inf, label = expression(bolditalic(&quot;rho = -0.524;p-value = 0.0002679 ***&quot;)), vjust = 1.5, hjust = 2.6, size = 4, face = &quot;balt&quot;)</code></pre>
<pre><code>## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
## ℹ Please use `linewidth` instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.</code></pre>
<pre><code>## Warning in annotate(&quot;text&quot;, x = Inf, y = Inf, label =
## expression(bolditalic(&quot;rho = -0.524;p-value = 0.0002679 ***&quot;)), : Ignoring
## unknown parameters: `face`</code></pre>
<pre><code>## `geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<pre><code>## Warning in is.na(x): is.na() applied to non-(list or vector) of type
## &#39;expression&#39;</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
<div id="generate-figure-4h" class="section level5">
<h5>Generate figure 4h:</h5>
<pre class="r"><code>#Spearman&#39;s rank correlation between average number of ε-end rings mantle hue:
cor.test(HPLC_mantle$Reflectance.average.Lambda.R50, HPLC_mantle$avg.epsilon.rings, method = &quot;spearman&quot;)</code></pre>
<pre><code>## 
##  Spearman&#39;s rank correlation rho
## 
## data:  HPLC_mantle$Reflectance.average.Lambda.R50 and HPLC_mantle$avg.epsilon.rings
## S = 19992, p-value = 0.03433
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##       rho 
## -0.316996</code></pre>
<pre class="r"><code># Make the figure
ggplot(HPLC_mantle, aes(x = Reflectance.average.Lambda.R50, y = avg.epsilon.rings)) +
  geom_point(aes(fill = Scientific.name), size = 2, shape = 21, color = &quot;black&quot;, alpha = 0.6) +
  geom_smooth(method = &quot;lm&quot;, se = F, color = &quot;black&quot;, size=0.5, linetype = &quot;dashed&quot;) + 
  scale_fill_manual(values = c(&quot;yellow&quot;, &quot;orange&quot;, &quot;red&quot;), name = NULL,
                    labels = c(expression(italic(&quot;D. benghalense&quot;)), &quot;Intermediates&quot;, expression(italic(&quot;D. psarodes&quot;)))) +
  xlab(&quot;Mantle hue (nm)&quot;) + 
  ylab(&quot;Epsilon rings&quot;) +
  ylim(c(0, 2.5))+
  theme_linedraw() +
  theme(
    panel.grid.major = element_line(color = &quot;grey90&quot;),
    panel.grid.minor = element_line(color = &quot;white&quot;),
    legend.position = &quot;top&quot;,
    plot.title = element_text(size = 8, face = &quot;bold&quot;)) + 
  annotate(&quot;text&quot;, x = Inf, y = Inf, label = expression(bolditalic(&quot;rho = -0.317;p-value = 0.03433 **&quot;)), vjust = 1.2, hjust = 2.6, size = 4)</code></pre>
<pre><code>## `geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<pre><code>## Warning in is.na(x): is.na() applied to non-(list or vector) of type
## &#39;expression&#39;</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
</div>
<div id="generate-figure-4i" class="section level5">
<h5>Generate figure 4i:</h5>
<pre class="r"><code># Spearman&#39;s rank correlation between average number C4(4’)-keto groups mantle hue:
cor.test(HPLC_mantle$Reflectance.average.Lambda.R50, HPLC_mantle$avg.4.keto_groups, method = &quot;spearman&quot;)</code></pre>
<pre><code>## Warning in cor.test.default(HPLC_mantle$Reflectance.average.Lambda.R50, :
## Cannot compute exact p-value with ties</code></pre>
<pre><code>## 
##  Spearman&#39;s rank correlation rho
## 
## data:  HPLC_mantle$Reflectance.average.Lambda.R50 and HPLC_mantle$avg.4.keto_groups
## S = 1012.7, p-value &lt; 2.2e-16
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##       rho 
## 0.9332841</code></pre>
<pre class="r"><code># Make the figure
ggplot(HPLC_mantle, aes(x = Reflectance.average.Lambda.R50, y = avg.4.keto_groups)) +
  geom_point(aes(fill = Scientific.name), size = 2, shape = 21, color = &quot;black&quot;, alpha = 0.6) +
  geom_smooth(method = &quot;loess&quot;, se = F, color = &quot;black&quot;, size=0.5, linetype = &quot;dashed&quot;) + 
  scale_fill_manual(values = c(&quot;yellow&quot;, &quot;orange&quot;, &quot;red&quot;), name = NULL,
                    labels = c(expression(italic(&quot;D. benghalense&quot;)), &quot;Intermediates&quot;, expression(italic(&quot;D. psarodes&quot;)))) +
  xlab(&quot;Mantle hue (nm)&quot;) +
  ylab(&quot;C4(4&#39;)-keto groups&quot;) +
  theme_linedraw() +
  theme(
    panel.grid.major = element_line(color = &quot;grey90&quot;),
    panel.grid.minor = element_line(color = &quot;white&quot;),
    legend.position = &quot;top&quot;,
    # axis.title.y = element_text(face = &quot;bold&quot;),
    # axis.title.x = element_text(face = &quot;bold&quot;),
    plot.title = element_text(size = 8, face = &quot;bold&quot;)) + 
  annotate(&quot;text&quot;, x = Inf, y = Inf, label = expression(bolditalic(&quot;rho = 0.933;p-value &lt; 2.2e-16 ****&quot;)), vjust = 1.5, hjust = 2.6, size = 4)</code></pre>
<pre><code>## `geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<pre><code>## Warning in is.na(x): is.na() applied to non-(list or vector) of type
## &#39;expression&#39;</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
</div>
</div>
<div id="figure-5a-gwas-manhattan-plot" class="section level4">
<h4>Figure 5a: GWAS Manhattan plot</h4>
<p>I conducted a Genome-Wide Association Study (GWAS) analysis using
GEMMA. This section demonstrates how to visualize the GEMMA output as a
Manhattan plot. For details on performing GWAS with GEMMA, please refer
to GEMMA_GAWS_pipeline.txt.</p>
<p>Load output file generated with GEMMA</p>
<pre class="r"><code>gwas &lt;- read.table(&quot;Data/DW.pre.filtered_n.106.mac3.biallelic.minDP3.minGQ25.NOindel.maxmiss0.4.BIMBAMimputedPC1untrnfm.lmm1.chronNumEdited.assoc.txt&quot;, header = T)</code></pre>
<p>Restructure the data to facilitate plotting.</p>
<pre class="r"><code>SNP &lt;- as.character(gwas$rs)
gwas.new &lt;- separate(gwas, col=rs, into = c(&#39;CHR&#39;, &#39;BP&#39;), sep = &#39;:&#39;)
CHR &lt;- as.integer(gwas.new$CHR)
BP &lt;- as.integer(gwas.new$BP)
P &lt;- as.numeric(gwas.new$p_wald)

# combine above vectors into a dataframe
gwas.clean &lt;- data.frame(SNP, CHR, BP, P)
head(gwas.clean)</code></pre>
<pre><code>##      SNP CHR  BP            P
## 1  47:47  47  47 5.356963e-01
## 2  33:69  33  69 4.707504e-06
## 3 47:107  47 107 6.132530e-01
## 4 47:126  47 126 4.245683e-01
## 5 47:139  47 139 2.303401e-01
## 6 17:149  17 149 8.934821e-01</code></pre>
<pre class="r"><code># Make teh QQ plots 
qq(gwas.clean$P)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<pre class="r"><code>plot(-log(gwas.clean$P)~gwas.clean$CHR) </code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-24-2.png" width="672" /></p>
<pre class="r"><code>hist(-log(gwas.clean$P))</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-24-3.png" width="672" /></p>
<p>Determine the FDR (False Discovery Rate) p-value threshold</p>
<pre class="r"><code># calculate FDR adjusted p-values
pval.corrected.fdr &lt;- p.adjust(P, method=&quot;fdr&quot;)
# make the data frame with adjusted p-valeus
gwas.clean &lt;- data.frame(SNP, CHR, BP, pval.corrected.fdr)
colnames(gwas.clean) &lt;- c(&quot;SNP&quot;, &quot;CHR&quot;, &quot;BP&quot;, &quot;P&quot;)
# get the SNPs that are significant according to FDR shreshold
significant_SNPs &lt;- subset(gwas.clean, pval.corrected.fdr &lt; 0.05)
# make the vector of SNPs to be colored on the plot
SNPs.to.highlight &lt;- significant_SNPs$SNP</code></pre>
<p>Generate the Manhattan plot</p>
<pre class="r"><code>manhattan(gwas.clean, main = &quot;Manhattan Plot with FDR p-value threshold&quot;,  cex = 0.6, cex.axis = 0.9, col = c(&quot;black&quot;, &quot;grey30&quot;), ylim=c(0,1.5), highlight = SNPs.to.highlight,  suggestiveline = F, genomewideline = -log10(0.05), chrlabs = c(1:44, &quot;W&quot;, &quot;Z&quot;, &quot;UN&quot;)) </code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
</div>
<div
id="figure-5b-genotypes-of-snps-significantly-associated-with-mantle-coloration"
class="section level4">
<h4>Figure 5b: Genotypes of SNPs significantly associated with mantle
coloration</h4>
<p>The genotype for each SNP was extracted from the .vcf file and saved
as a .csv file.</p>
<p>Load the data</p>
<pre class="r"><code>geno.data &lt;- read.csv(&quot;Data/Genotype_data-GWAS-SNP.csv&quot;, stringsAsFactors = T)</code></pre>
<p>Reshaping the data</p>
<pre class="r"><code>geno_long &lt;- geno.data %&gt;%
  pivot_longer(cols = c(Chr4_17938638,Chr11_30318410,Chr31_3082739,Chr31_3082775,Chr31_3082777),
               names_to = &quot;Variable&quot;,
               values_to = &quot;Values&quot;) %&gt;%
  filter(Values != &quot;-1&quot;) # to remove missing data
geno_long$Values &lt;- as.character(geno_long$Values)
geno_long$Variable &lt;- factor(geno_long$Variable, levels = c(&quot;Chr4_17938638&quot;,  &quot;Chr11_30318410&quot;, &quot;Chr31_3082739&quot;,  &quot;Chr31_3082775&quot;,&quot;Chr31_3082777&quot;)) </code></pre>
<p>Generate the plots</p>
<pre class="r"><code>ggplot(geno_long, aes(fill = Group_phenotype, y = LAB_pca_PC1, x = Values)) +
  geom_jitter(position = position_jitter(0.3), size = 2, shape= 21, color = &quot;black&quot;, alpha = 0.6) +
  scale_fill_manual(values = c( &quot;orange&quot;, &quot;red&quot;, &quot;red&quot;, &quot;yellow&quot;, &quot;yellow&quot;, &quot;yellow&quot;)) + 
  facet_wrap(~Variable, scales = &quot;fixed&quot;, labeller = as_labeller(c(&quot;Chr4_17938638&quot;=&quot;Chromosome 4: 17938638&quot;,
                                                                  &quot;Chr11_30318410&quot;=&quot;Chromosome 11: 30318410&quot;,
                                                                  &quot;Chr31_3082739&quot;=&quot;Chromosome 31: 3082739&quot;,
                                                                  &quot;Chr31_3082775&quot;=&quot;Chromosome 31: 3082775&quot;,
                                                                  &quot;Chr31_3082777&quot;=&quot;Chromosome 31: 3082777&quot;))) +  
  xlab(&quot;Genotype&quot;) + 
  ylab(&quot;Phenotype score (PC1 of LAB PCA)&quot;) + 
  theme_linedraw() +
  theme(
    panel.grid.major = element_line(color = &quot;grey70&quot;),
    panel.grid.minor = element_line(color = &quot;white&quot;),
    strip.background = element_rect(color = &quot;grey50&quot;, fill = &quot;grey50&quot;),
    strip.text = element_text(face = &quot;bold&quot;),
    legend.position = &quot;none&quot;,
    axis.title.y = element_text(size = 9)) </code></pre>
<pre><code>## Warning: Removed 15 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
</div>
</div>
<div id="statistical-analysis" class="section level3">
<h3>Statistical Analysis</h3>
<div
id="calculate-the-mean-hue-and-ci-of-the-mean-hue-for-crown-and-mantle-feathers"
class="section level4">
<h4>Calculate the mean hue and CI of the mean hue for crown and mantle
feathers</h4>
<pre class="r"><code># Read the data 
HPLC &lt;- read.csv(&quot;Data/HPLC_data_Mantle_Crown_feathers_All_Diniopium.csv&quot;, stringsAsFactors = T)
# Calculation
summary_stats &lt;- HPLC %&gt;%
  group_by(feathers, Species) %&gt;%
  summarize(
    mean_lambda_R50 = mean(Reflectance.average.Lambda.R50, na.rm = TRUE),
    ci_lower = mean_lambda_R50 - qt(0.975, length(Reflectance.average.Lambda.R50) - 1) * (sd(Reflectance.average.Lambda.R50, na.rm = TRUE) / sqrt(length(Reflectance.average.Lambda.R50))),
    ci_upper = mean_lambda_R50 + qt(0.975, length(Reflectance.average.Lambda.R50) - 1) * (sd(Reflectance.average.Lambda.R50, na.rm = TRUE) / sqrt(length(Reflectance.average.Lambda.R50))),
    ci_lower = round(ci_lower, 2),
    ci_upper = round(ci_upper, 2))</code></pre>
<pre><code>## `summarise()` has grouped output by &#39;feathers&#39;. You can override using the
## `.groups` argument.</code></pre>
<pre class="r"><code># View the data
summary_stats</code></pre>
<pre><code>## # A tibble: 6 × 5
## # Groups:   feathers [2]
##   feathers Species        mean_lambda_R50 ci_lower ci_upper
##   &lt;fct&gt;    &lt;fct&gt;                    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 Crown    D. benghalense            598.     596.     600.
## 2 Crown    D. psarodes               600.     597.     603.
## 3 Crown    Intermediates             598.     578.     619.
## 4 Mantle   D. benghalense            545.     541.     550.
## 5 Mantle   D. psarodes               602.     599.     604.
## 6 Mantle   Intermediates             575.     567.     583.</code></pre>
<!-- ### Supplementary Figures -->
<!-- Please note that Ranasinghe et al., 2025 have plotted the hue values (H3) that were manually calculated from the reflectance data, rather than the H3 values generated by _pavo_. Load the updated data files with the new hue (H3) values and generate the plots. -->
<!-- ```{r} -->
<!-- # Read the edited crown data -->
<!-- colorimatrics_crown <- read.csv("Data/Colorimatric.Data-Dinopium_Crown_from_pavo_Edited.csv", stringsAsFactors = T) -->
<!-- # Mutate the data  -->
<!-- Data_long <- colorimatrics_crown %>% -->
<!--   pivot_longer(cols = c(S9, H3, B2), -->
<!--                names_to = "Variable", -->
<!--                values_to = "Values") -->
<!-- levels(Data_long$Species) # To get the categories in Species column -->
<!-- Data_long$Species <- factor(Data_long$Species, levels = c("D.benghalense", "hybrids",  "D.psarodes")) -->
<!-- Data_long$Variable <- factor(Data_long$Variable, levels = c("B2","S9", "H3"))  -->
<!-- # make Figures 2b–d -->
<!-- ggplot(Data_long, aes(y = Values, x = Species, fill=Species)) + -->
<!--   geom_jitter(position = position_jitter(0.3), size = 2.5, shape = 21, color = "black", alpha = 0.6) + -->
<!--   stat_summary(fun.data = "mean_cl_normal",  fun.args = list(mult = 2.5),  -->
<!--                geom = "pointrange",  size = 0.3, shape = 23, fill = "black")+ -->
<!--   facet_wrap(~Variable, ncol = 4, scales = "free", -->
<!--              labeller = labeller(Variable=c("B2"="Mean brightness", -->
<!--                                             "S9"="Carotenoid Chroma", -->
<!--                                             "H3"= "Mean hue"))) + -->
<!--   scale_fill_manual(values = c("yellow", "orange", "red"), -->
<!--                     name = NULL, -->
<!--                     labels = c(expression(paste("Yellow ", italic("(D. benghalense)"))),  -->
<!--                                 "Intermediate phenotype",  -->
<!--                                expression(paste("Red ", italic("(D. psarodes)"))))) + -->
<!--   ylab("Intensity") + -->
<!--   xlab("Phenotypic groups")+ -->
<!--   scale_x_discrete(labels = NULL) + -->
<!--   theme_linedraw() + -->
<!--   theme(aspect.ratio = 1, -->
<!--     panel.grid.major = element_line(color = "grey90"), -->
<!--     panel.grid.minor = element_line(color = "white"), -->
<!--     strip.background = element_rect(color = "grey50", fill = "grey50"),   -->
<!--     strip.text = element_text(face = "bold", size = 11, -->
<!--                               margin = margin(10, 10, 10, 10)), -->
<!--     legend.text = element_text(size = 11), -->
<!--     panel.spacing = unit(1.5, "lines"), -->
<!--     legend.position = "top", -->
<!--     axis.title.y = element_text(face = "bold"), -->
<!--     axis.title.x = element_text(face = "bold")) + -->
<!--   ggtitle("Colorimatric variables of crown feathers")  -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # Read the edited mantle data -->
<!-- colorimatrics_mantle <- read.csv("Data/Colorimatric.Data-Dinopium_mantle_from_pavo_Edited.csv", stringsAsFactors = T) -->
<!-- # Mutate the data  -->
<!-- Data_long <- colorimatrics_mantle %>% -->
<!--   pivot_longer(cols = c(S9, H3, B2), -->
<!--                names_to = "Variable", -->
<!--                values_to = "Values") -->
<!-- Data_long$Species <- factor(Data_long$Species, levels = c("D. benghalense", "Hybrid",  "D. psarodes")) -->
<!-- Data_long$Variable <- factor(Data_long$Variable, levels = c("B2","S9", "H3"))  -->
<!-- # make Figures 2f–g -->
<!-- ggplot(Data_long, aes(y = Values, x = Species, fill=Species)) + -->
<!--   geom_jitter(position = position_jitter(0.3), size = 2.5, shape = 21, color = "black", alpha = 0.6) + -->
<!--   stat_summary(fun.data = "mean_cl_normal",  fun.args = list(mult = 2.5),  -->
<!--                geom = "pointrange",  size = 0.3, shape = 23, fill = "black")+ -->
<!--   facet_wrap(~Variable, ncol = 4, scales = "free", -->
<!--              labeller = labeller(Variable=c("B2"="Mean brightness", -->
<!--                                             "S9"="Carotenoid Chroma", -->
<!--                                             "H3"= "Mean hue"))) + -->
<!--   scale_fill_manual(values = c("yellow", "orange", "red"), -->
<!--                     name = NULL, -->
<!--                     labels = c(expression(paste("Yellow ", italic("(D. benghalense)"))),  -->
<!--                                 "Intermediate phenotype",  -->
<!--                                expression(paste("Red ", italic("(D. psarodes)"))))) + -->
<!--   ylab("Intensity") + -->
<!--   xlab("Phenotypic groups")+ -->
<!--   scale_x_discrete(labels = NULL) + -->
<!--   theme_linedraw() + -->
<!--   theme(aspect.ratio = 1, -->
<!--     panel.grid.major = element_line(color = "grey90"), -->
<!--     panel.grid.minor = element_line(color = "white"), -->
<!--     strip.background = element_rect(color = "grey50", fill = "grey50"),   -->
<!--     strip.text = element_text(face = "bold", size = 11, -->
<!--                               margin = margin(10, 10, 10, 10)), -->
<!--     legend.text = element_text(size = 11), -->
<!--     panel.spacing = unit(1.5, "lines"), -->
<!--     legend.position = "top", -->
<!--     axis.title.y = element_text(face = "bold"), -->
<!--     axis.title.x = element_text(face = "bold")) + -->
<!--   ggtitle("Colorimatric variables of mantle feathers")  -->
<!-- ``` -->
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5,h6",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
